### PR TITLE
Fix ProjectData cache key and update db entities with enum.

### PR DESCRIPTION
**Pull request by gkillough**
There are virtually no functional changes. I refactored to switch from Strings to enums in a few database objects.

### CODE REVIEW

File: src/main/java/com/blackducksoftware/integration/hub/alert/digest/model/ProjectDataFactory.java [hunk 1]
```diff
@@ -50,17 +50,17 @@
     public Collection<ProjectData> createProjectDataCollection(final Collection<NotificationEntity> notifications, final DigestTypeEnum digestType) {
         final Map<String, ProjectDataBuilder> projectDataMap = new LinkedHashMap<>();
         for (final NotificationEntity entity : notifications) {
-            final String projectKey = entity.getEventKey();
+            final String projectKey = entity.getProjectName() + entity.getProjectVersion();
```
Comment (gkillough) [inline src/main/java/com/blackducksoftware/integration/hub/alert/digest/model/ProjectDataFactory.java]: This is the fix.

File: src/main/java/com/blackducksoftware/integration/hub/alert/digest/DigestRemovalProcessor.java [hunk 2]
```diff
@@ -74,7 +74,7 @@ private String createCacheKey(final NotificationEntity entity) {
     }
 
     private boolean processPolicyNotifications(final Map<String, NotificationEntity> categoryMap, final NotificationEntity entity) {
-        final String notificationType = entity.getNotificationType();
+        final String notificationType = entity.getNotificationType().name();
         if (NotificationCategoryEnum.POLICY_VIOLATION.name().equals(notificationType)) {
```
Comment (bamandel) [inline src/main/java/com/blackducksoftware/integration/hub/alert/digest/DigestRemovalProcessor.java]: You can compare just enums and remove all string related expressions here

File: src/main/java/com/blackducksoftware/integration/hub/alert/digest/DigestRemovalProcessor.java [hunk 3]
```diff
@@ -93,7 +93,7 @@ private boolean processPolicyNotifications(final Map<String, NotificationEntity>
     }
 
     private boolean processVulnerabilityNotifications(final String cacheKey, final Map<String, NotificationEntity> categoryMap, final NotificationEntity entity) {
-        final String notificationType = entity.getNotificationType();
+        final String notificationType = entity.getNotificationType().name();
```
Comment (bamandel) [inline src/main/java/com/blackducksoftware/integration/hub/alert/digest/DigestRemovalProcessor.java]: You can also try getting just the enum here as well but would require some additional changes to the maps below

File: src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/NotificationEntityTest.java [hunk 4]
```diff
@@ -72,7 +76,27 @@ public void assertEntityFieldsFull(final NotificationEntity entity) {
...
+    @Test
+    public void testEntity() throws JSONException {
+        // TODO figure out why the hash code keeps changing
+        final NotificationEntity configEntity = getMockUtil().createEntity();
+
+        assertEntityFieldsFull(configEntity);
+        assertEquals(Long.valueOf(getMockUtil().getId()), configEntity.getId());
+
+        // TODO figure out why the hash code keeps changing
+        // final int configHash = configEntity.hashCode();
```
Comment (bamandel) [inline src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/NotificationEntityTest.java]: Looks like duplicate TODOs here

File: src/main/java/com/blackducksoftware/integration/hub/alert/digest/DigestRemovalProcessor.java (hunk 2 above)
Comment (gkillough) [reply src/main/java/com/blackducksoftware/integration/hub/alert/digest/DigestRemovalProcessor.java]: That would require modifying the entity cache stuff. We don't want to do that for now.

File: src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/NotificationEntityTest.java (hunk 4 above)
Comment (gkillough) [reply src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/NotificationEntityTest.java]: Good catch

File: src/main/java/com/blackducksoftware/integration/hub/alert/digest/DigestRemovalProcessor.java (hunk 2 above)
Comment (gkillough) [reply src/main/java/com/blackducksoftware/integration/hub/alert/digest/DigestRemovalProcessor.java]: Actually, I think there's an easy way to do it.

**Pull request merged**